chain/get_ancestors.go

```diff
@@ -51,37 +51,32 @@ func GetRecentAncestors(ctx context.Context, base types.TipSet, provider TipSetP
 		earliestAncestorHeight = types.NewBlockHeight(uint64(0))
 	}
 
-	// Step 1 -- gather all tipsets with a height greater than the earliest
-	// possible proving period start still in scope for the given head.
 	iterator := IterAncestors(ctx, provider, base)
+	// Step 1 -- gather all tipsets up to the first tipset with a height less than
+	// or equal to the earliest possible proving period start
 	provingPeriodAncestors, err := CollectTipSetsOfHeightAtLeast(ctx, iterator, earliestAncestorHeight)
 	if err != nil {
 		return nil, err
 	}
-	firstExtraRandomnessAncestorsCids, err := provingPeriodAncestors[len(provingPeriodAncestors)-1].Parents()
-	if err != nil {
-		return nil, err
-	}
-	// no parents means hit genesis so return the whole chain
-	if firstExtraRandomnessAncestorsCids.Len() == 0 {
-		return provingPeriodAncestors, nil
-	}
 
-	// Step 2 -- gather the lookback tipsets directly preceding provingPeriodAncestors.
-	lookBackTS, err := provider.GetTipSet(firstExtraRandomnessAncestorsCids)
-	if err != nil {
-		return nil, err
+	// Step 2 -- skip the next tipset which added if iterator not complete
```

Comment:
nit: it would be helpful to explain that the iterator is currently pointing at the lowest height element of `provingPeriodAncestors` and needs to be updated to avoid duplicates.